packages/dds/tree/src/feature-libraries/treeTextFormat.ts

```diff
@@ -0,0 +1,225 @@
+ * Licensed under the MIT License.
+ */
+
+import {
+    ITreeCursor,
+    TreeNavigationResult,
+    mapCursorField,
+} from "../forest";
+import { TreeSchemaIdentifier } from "../schema";
+import {
+    FieldKey,
+    TreeType,
+    TreeValue,
+    Value,
+} from "../tree";
+
+/**
+ * This modules provides support for reading and writing a human readable (and editable) tree format.
+ *
+ * This implementation can handle all trees (so it does not need a fallback for any special cases),
+ * and is not optimized.
+ *
+ * It's suitable for testing and debugging,
+ * though it could also reasonably be used as a fallback for edge cases or for small trees.
+ *
+ * The serialized format is valid utf-8, and also includes a json compatible intermediate in memory format.
+ *
+ * This format is currently not stable: its internal contents are not considered public APIs and may change.
+ * There is currently no guarantee that data serialized with this library will
+ * be loadable with a different version of this library.
+ *
+ * TODO: stabilize this format (probably after schema are more stable).
+ *
+ * This format does not include schema: typically schema would be stored alongside data in this format.
+ *
+ * TODO: Use placeholders.
+ * build / add operations should be able to include detached ranges instead of children directly.
+ * summaries should be able to reference unloaded chunks instead of having children directly.
+ * Leverage placeholders in the types below to accomplish this.
+ * Determine how this relates to Cursor: should cursor be generic over placeholder values?
+ * (Could use them for errors to allow non erroring cursors?)
+ *
+ * Note:
+ * Currently a lot of Tree's codebase is using json for serialization.
+ * Because putting json strings inside json works poorly (adds lots of escaping),
+ * for now this library actually outputs and inputs the Json compatible type PlaceholderTree
+ * rather than actual strings.
+ */
+
+/**
+ * Json compatible map as object.
+ * Keys are TraitLabels,
+ * Values are the content of the trait specified by the key.
+ * @public
+ */
+ export interface FieldMap<TChild> {
+    [key: string]: TreeNodeSequence<TChild>;
+}
+
+/**
+ * A sequence of Nodes that make up a trait under a Node
+ * @public
+ */
+export type TreeNodeSequence<TChild> = readonly TChild[];
+
+/**
+ * An object which may have traits with children of the given type underneath it
+ * @public
+ */
+export interface WithFields<TChild> {
+    fields?: Readonly<FieldMap<TChild>>;
+}
+
+/**
+ * The fields required by a node in a tree
+ * @public
+ */
+export interface NodeData {
+    /**
+     * A payload of arbitrary serializable data
+     */
+    value?: TreeValue;
+
+    /**
+     * The meaning of this node.
+     * Provides contexts/semantics for this node and its content.
+     * Typically use to associate a node with metadata (including a schema) and source code (types, behaviors, etc).
+     */
+    readonly type: TreeSchemaIdentifier;
```

Comment:
Treee uses structural identity, which is captured by the shape of the tree.

If you instead mean the optional not yet implemented identifier-based identities, they will be either:
- Stored on trees using the existing functionality (ex: store the identifier in an identifier field).
- Have some API added special for them in all our tree abstractions (ex: cursor, forest etc).

Either way, it does not make sense to do anything special for them in in the storage format currently, since it will either just wrk (in the first case) or we lack the APIs to encode or decode them (in the second).